torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -114,7 +114,7 @@ static Variable valueToTensor(const at::Type & type, PyObject* value) {
   if (THPVariable_Check(value)) {
     return reinterpret_cast<THPVariable*>(value)->cdata;
   }
-  if (THPUtils_checkLong(value)) {
+  if (THPUtils_checkLong(value) || PyBool_Check(value)) {
```

Comment:
But now I wonder, does this mean `THPUtils_checkLong` should accept bool? Because `THPUtils_unpackLong` works with bool (apparently). Do you know? (This bandaid is OK for this PR, but it would be good to sort this question out.)

Reply:
I would say that THPUtils_checkLong should not accept bool as it gives some more flexibility in terms of restrictions if that makes sense. There might be situations when we want long but not bool.

Reply:
Well, under your reasoning, I might claim that `THPUtils_unpackLong` shouldn't accept bool ;) (And conversely, there might be situations where we want both; it might be productive to prophylactically audit these sites to see if there are other cases where we should accept bool too.)

Reply:
that's not really the heart of the issue, it's that the check and the unpack aren't symmetric and there isn't any indication as to why.  If you look at the code, it's because unpackLong calls a python function that converts to long (so bool works, but also a lot of other stuff).  Maybe this is correct (because e.g. you want check to be really fast), but it's not explained.

Reply:
@ezyang, fair enough. ok, i will take a look into this and will follow up on this.

Reply:
They aren't symmetric and aren't intended to be, just like the Python C API:

`PyLong_Check` is true only for `PyLongObject` and subtypes
`PyLong_AsLongAndOverflow` converts the object to a `PyLongObject` and then returns the value.

The general practice in Python is **not** to check the object before converting. PyTorch has had to deviate from this practice originally because of ambiguous overloads inherited from Lua torch

Reply:
OK, I stand corrected!

Reply:
@izdeby can you incorporate @colesbury's comment into the code in a reasonable place?

Reply:
Sure, will do.